Refactor timers in `Fullnode` and `Validator`

hewison-chris opened the pull request:
Rather than using a dynamic array to store the timers an AA is used with
the key an integer. The timers are then accessed using enums defined in
Fullnode and Validator to make sure the right timer is accessed.

hewison-chris committed:
Refactor timers in Fullnode and Validator

Rather than using a dynamic array to store the timers an AA is used with
the key an integer. The timers are then accessed using enums defined in
Fullnode and Validator to make sure the right timer is accessed.

linked0 reviewed (approved):
LGTM

omerfirmak commented:
This was a much needed change, thanks! But CI does not like it.

Geod24 commented on source/agora/node/Validator.d:
```diff
@@ -61,6 +61,13 @@ import core.time;
 {
+    protected enum ValidatorTimers
+    {
+        NameRegistration = super.FullNodeTimers.max + 1,
```
Don't need the `super` here.

hewison-chris replied:
Thought it made it clearer (like using `this` when relevant but not required) but can remove.

hewison-chris committed:
Temporary sleeps in ValidaorCleanRestart test

We have issue Agora #2971 `Frequent failure on ValidatorCleanRestart and
RestoreSlashingInfo` to address this. Currently this test prevents us
moving forward with changes that will help us make it more stable. Once
we have the other changes in place we can remove these sleeps and fix.

hewison-chris commented:
Rebased on latest.

hewison-chris commented:
Included in https://github.com/bosagora/agora/pull/3000/commits which is green so will close this one.
